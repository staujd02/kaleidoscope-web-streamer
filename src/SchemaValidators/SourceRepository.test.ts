import React from 'react';
import SourceRepositoryValidation from './SourceRepository';
import SourceValidation from './Source';

describe('The Source Repository Validator', () => {

    let repoState: SourceRepository;

    beforeEach(() => {
        repoState = {
            streams: [
                {
                    title: "A title",
                    duration: 5000,
                    isEnabled: false,
                    source: 'www.somewhere.com',
                    sortOrder: 3,
                    key: 1
                }
            ]
        }
    });

    it('should validate a correct repository', () => {
        spyOn(SourceValidation, 'validate').and.callFake(() => {});
        expect(() => SourceRepositoryValidation.validate(repoState)).not.toThrowError();
    });

    describe('when given an invalid repository', () => {

        describe('given that repo is missing a stream list', () => {

            beforeEach(() => {
                (repoState.streams as any) = undefined;
            });

            it('rejects it', () => {
                expect(() => SourceRepositoryValidation.validate(repoState)).toThrowErrorMatchingSnapshot();
            });
        });
        
        describe('given that repo has a stream that is invalid', () => {

            beforeEach(() => {
                spyOn(SourceValidation, 'validate').and.throwError("Fake my error");
            });

            it('rejects it', () => {
                expect(() => SourceRepositoryValidation.validate(repoState)).toThrowErrorMatchingSnapshot();
            });
        });
    });
});