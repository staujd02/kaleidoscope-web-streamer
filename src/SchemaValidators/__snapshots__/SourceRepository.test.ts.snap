// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Source Repository Validator when given an invalid repository given that repo has a stream that is invalid rejects it 1`] = `"Fake my error"`;

exports[`The Source Repository Validator when given an invalid repository given that repo is missing a stream list rejects it 1`] = `"Missing Property Exception: 'streams' is not defined"`;
